app/routines/newflow/verify_educator.rb

```diff
@@ -0,0 +1,88 @@
+module Newflow
+  # Verify their Educator status given their (SheerID) Verification ID
+  # Expects a user to already exist with that SheerID Verification ID
+  class VerifyEducator
+    lev_routine active_job_enqueue_options: { queue: :educator_signup_queue }, use_jobba: true
+    uses_routine UpsertSheeridVerification
+    uses_routine UpdateSalesforceLead
+
+    protected ###############
+
+    def exec(verification_id:, user:)
+      status.set_job_name(self.class.name)
+      status.set_job_args(verification_id: verification_id.to_s, user: user.to_global_id.to_s)
+
+      verification_record = fetch_verification(verification_id)
+      transfer_errors_from(verification_record, {type: :verbatim}, :fail_if_errors)
+      return if !verification_record.verified?
+
+      # If the user is already faculty verified, nothing to do. If they're a student, don't faculty verify.
+      return if user.confirmed_faculty? || user.student?
+
+      email = EmailAddress.verified.find_by(value: verification_record.email)
+
+      capture_mismatch_error!(verification_id, email, user) and return if email_mismatch?(user, email)
+
+      if update_user(user, verification_record) && update_salesforce_lead_for(user)
+        log_success(verification_id, user)
+      else
+        handle_error(verification_id, user)
+      end
+    end
+
+    private #################
+
+    def update_user(user, verification)
+      user.update(
+        first_name: verification.first_name,
+        last_name: verification.last_name,
+        sheerid_reported_school: verification.organization_name,
+        faculty_status: verification.current_step_to_faculty_status,
+        sheerid_verification_id: verification.verification_id
+      )
+    end
+
+    def fetch_verification(verification_id)
+      @fetch_verification ||= run(UpsertSheeridVerification, verification_id: verification_id).outputs.verification
+    end
+
+    def update_salesforce_lead_for(user)
+      run(UpdateSalesforceLead, user: user).outputs.lead.errors.none?
+    end
+
+    def email_mismatch?(user, email)
+      user.blank? || email.blank? || email.user_id != user.id
```

Comment:
this could all be simplified to 
```
email&.user_id != user&.id
```

Reply:
If both are `nil` then it would still evaluate to true, so that's why I decided to make it this way. But if the rest of the logic is correct, then user should never be nil. In that case, you might be right. Fixed.